src/osd/ReplicatedPG.cc

```diff
@@ -3904,7 +3904,7 @@ int ReplicatedPG::do_writesame(OpContext *ctx, OSDOp& osd_op)
   }
 
   while (write_length) {
-    write_op.indata.append(osd_op.indata.c_str(), op.writesame.data_length);
+    write_op.indata.append(osd_op.indata);
```

Comment:
on client side `Objecter::prepare_writesame_op()`, we can assume that they are equal. but can we assert `op.indata.length() == op.writesame.data_length` on server side?